File: sentimental-cash/cash.py
from cs50 import get_float

def main():
    # Ask for input until user provides a valid positive number
    while True:
        try:
            cents = get_float("Change owed: ")
            if cents > 0:
                break
        except ValueError:
            # If the input is not numeric, ask again
            continue

    # Convert dollars to cents and round to avoid floating-point errors
    cents = round(cents * 100)

    coins = 0

    # Calculate coins
    coins += cents // 25
    cents %= 25

    coins += cents // 10
    cents %= 10

    coins += cents // 5
    cents %= 5

    coins += cents

    print(coins)

# Run the program
if __name__ == "__main__":
    main()

File: cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    int cent;
    do
    {
        cent = get_float("Change Owed: "); // We ask the user to introduce a number must be positive
        if (cent < 0)
        {
            printf("Introduce a positve number\n"); // If we introduce a non-valid numer message will apear
        }
    }
    while (cent < 0);

    int coins = 0;

    coins += cent / 25;
    cent %= 25;

    coins += cent / 10;
    cent %= 10;

    coins += cent / 5;
    cent %= 5;

    coins += cent;

    printf("%i\n", coins);
}
